### PR TITLE
remove bag name logic

### LINKED ISSUE #600: Remove bag name logic
Currently the bag is sometimes prefixed by the translated word for bag

Here is how it currently works: https://github.com/the-clothing-loop/website/blob/ece43145501884e76001fe4d7d6a0683e37312ac/server/docs/bag_name.md

- [x] **Remove that bag logic**
  - [x] From the app
  - [x] From the website
  - [x] From the server
- [x] **Remove the above mentioned documentation**
- [x] **Add a migration sql script to be run so that the bags Look like they are the same:**
      From `{ translated_text_bag} Name`, find the language used for that loop, and replace with `Bag Name`

---
Here are the rules for the migration script:

1. The bags that have a name shorter than 7 characters will get a `Bag ` (in a translated form) prepended.
2. To figure out which language to translate it to, the related Loop and Hosts of said Loop must be retrieved:
   1. As there can be multiple hosts in one loop (often one being more active than the other) pick the host that contains a selected language, and set it as such.
   2. If the no host has set a language, default to `nl`.

An Error correction script must be added to find: 
1. all Bags with a `nl` prefix
2. and where its Loop country is not `nl` or `be`

This is to be run post migration to fix the handful of non-dutch loops

### PR DESCRIPTION
**Pull request by tobifaf**
resolves #600

### CODE REVIEW

File: server/sql/prepend-i18n-bag-name.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+UPDATE
+    bags b
+        JOIN user_chains uc ON
+        b.user_chain_id = uc.id
+        JOIN users u ON
```
Comment (lil5) [inline server/sql/prepend-i18n-bag-name.sql:5]: This looks like it works but what if the bag was created by a loop host that was Dutch and hosts a Dutch loop, but the current bag holder is English or French?

A sub query is needed to get the user ID of a host of the bags loop and get the i18n for them.

**Review by lil5 — CHANGES_REQUESTED** [review]

File: server/sql/prepend-i18n-bag-name.sql (hunk 1 above)
Comment (tobifaf) [reply server/sql/prepend-i18n-bag-name.sql:5]: okay, that's also ok. Do I just get the host with the lowest ID?

Comment (lil5) [reply server/sql/prepend-i18n-bag-name.sql:5]: any host with a i18n that contains 2 letters, default to dutch if not.

Comment (tobifaf) [reply server/sql/prepend-i18n-bag-name.sql]: You got me there :) I misinterpreted the fields. Pushed a fix, but I assumed you mean default to English?

File: server/sql/prepend-i18n-bag-name.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+UPDATE bags b
+    JOIN user_chains bag_chain ON 
+        b.user_chain_id = bag_chain.id
+    JOIN user_chains uc ON
+        bag_chain.chain_id = uc.chain_id AND uc.is_chain_admin = 1
+    JOIN users u ON
+        uc.user_id = u.id AND LENGTH(u.i18n) = 2
```
Comment (lil5) [inline server/sql/prepend-i18n-bag-name.sql:7]: Correct me if I'm wrong but wouldn't this have the potential for it to be overwritten?

Loop x has two hosts A ( i18n: en ) & B ( i18n: empty )

The query finds bag K,and gives a list of;
1. Bag K, host A, en
2. Bag K, host B, empty

The SET would change the bag name first to "Bag K" then to "Tas K". 

If so use a sub query to find the first host with an actionable i18n (not "nl" or empty)

**Review by lil5 — CHANGES_REQUESTED** [review]

File: server/sql/prepend-i18n-bag-name.sql (hunk 1 above)
Comment (lil5) [reply server/sql/prepend-i18n-bag-name.sql]: Nope default to Dutch, damage control will be significantly easier

Comment (tobifaf) [reply server/sql/prepend-i18n-bag-name.sql]: done

File: server/sql/prepend-i18n-bag-name.sql (hunk 2 above)
Comment (tobifaf) [reply server/sql/prepend-i18n-bag-name.sql]: Fixed (i hope). The query is still looping over all hosts, but will only change the number the first time. (order: nl, fr, en, others=nl)

**Comment by lil5** [comment]
The sql script is getting to complex the review from my phone I'll need to test it at some time 🥸

I'll mark the in review and leave it for November

**Comment by NinavanBruggen** [comment]
Hi!,

So I might not totally understand the complexity of the problem. So forgive me if so.

I'm also not totally sure if also the bags with more than 7 characters should include the word 'bag' from now on.
If so I think those should be updated first:
`UPDATE bags 

SET number = IF(
                LENGTH(number) > 7, 
number = CONCAT('Bag ', number);`

Then all the Dutch and French bags can be updated. If the field already contains the word 'Bag' it should ignore it. 

`UPDATE bags 
                
SET number = REPLACE(number, 'Tas', 'Bag') 
SET number = REPLACE(number, 'Sac', 'Bag')`

On another note: I think bag.name is a better field name than bag.number since it contains strings now.

**Comment by tobifaf** [comment]
> Hi!,
> 
> So I might not totally understand the complexity of the problem. So forgive me if so.
> 
> I'm also not totally sure if also the bags with more than 7 characters should include the word 'bag' from now on. If so I think those should be updated first:
> 
> ```
> UPDATE bags 
> 
> SET number = IF(
>                 LENGTH(number) > 7, 
> number = CONCAT('Bag ', number);
> ```
> 
> Then all the Dutch and French bags can be updated. If the field already contains the word 'Bag' it should ignore it.
> 
> ```
> UPDATE bags 
>                 
> SET number = REPLACE(number, 'Tas', 'Bag') 
> SET number = REPLACE(number, 'Sac', 'Bag')
> ```
> 
> On another note: I think bag.name is a better field name than bag.number since it contains strings now.

Thanks for your input!

Bags with more than 7 characters should not be updated. Here's a small diagram of the current way it works: https://github.com/the-clothing-loop/website/blob/ece43145501884e76001fe4d7d6a0683e37312ac/server/docs/bag_name.md

I agree the field name number is not very clear.

**Comment by lil5** [comment]
Originally the field was used for numbers only, it was later changed to be used for text as well. This migration is the perfect moment to alter the column name (to `name`)!

**Comment by NinavanBruggen** [comment]
> > Hi!,
> > So I might not totally understand the complexity of the problem. So forgive me if so.
> > I'm also not totally sure if also the bags with more than 7 characters should include the word 'bag' from now on. If so I think those should be updated first:
> > ```
> > UPDATE bags 
> > 
> > SET number = IF(
> >                 LENGTH(number) > 7, 
> > number = CONCAT('Bag ', number);
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Then all the Dutch and French bags can be updated. If the field already contains the word 'Bag' it should ignore it.
> > ```
> > UPDATE bags 
> >                 
> > SET number = REPLACE(number, 'Tas', 'Bag') 
> > SET number = REPLACE(number, 'Sac', 'Bag')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > On another note: I think bag.name is a better field name than bag.number since it contains strings now.
> 
> Thanks for your input!
> 
> Bags with more than 7 characters should not be updated. Here's a small diagram of the current way it works: https://github.com/the-clothing-loop/website/blob/ece43145501884e76001fe4d7d6a0683e37312ac/server/docs/bag_name.md
> 
> I agree the field name number is not very clear.

Ok well in that case the code should be:

```
UPDATE bags 
SET number = CASE WHEN LENGTH(number) < 7 THEN
      REPLACE(number, 'Tas', 'Bag')
  END

````
and
````
  
UPDATE bags 
SET number = CASE WHEN LENGTH(number) < 7 THEN
      REPLACE(number, 'Sac', 'Bag')
  END
```

And then change the field name to 'name'. 

I have added the if statement to avoid 'spooktas' to be changed to 'spookbag'. 
Unfortunately I haven't found a way to do two REPLACE statements in one UPDATE statement. 
Hope this helps.

**Comment by lil5** [comment]
I just added some e2e tests and the script works flawlessly

```go
		{
			OriginalBagName: "one",
			I18nHost1:       "",
			I18nHost2:       "fr",
			ExpectedBagName: "Sac one",
		},
		{
			OriginalBagName: "lorem ipsum",
			I18nHost1:       "",
			I18nHost2:       "fr",
			ExpectedBagName: "lorem ipsum",
		},
		{
			OriginalBagName: "one",
			I18nHost1:       "en",
			I18nHost2:       "fr",
			ExpectedBagName: "Sac one",
		},
		{
			OriginalBagName: "one",
			I18nHost1:       "fr",
			I18nHost2:       "",
			ExpectedBagName: "Sac one",
		},
		{
			OriginalBagName: "one",
			I18nHost1:       "fr",
			I18nHost2:       "en",
			ExpectedBagName: "Sac one",
		},
		{
			OriginalBagName: "one",
			I18nHost1:       "",
			I18nHost2:       "",
			ExpectedBagName: "Tas one",
		}, {
			OriginalBagName: "one",
			I18nHost1:       "",
			I18nHost2:       "",
			ExpectedBagName: "Tas one",
		},
```

If there are any edge cases I forgot I'd be happy to have a look.
The bags with the prefix from a loop that is located outside of the Netherlands I'll fix after the migration (error correction script)

**Pull request merged**